qualtran/bloqs/qsp/generalized_qsp.py

```diff
@@ -338,12 +342,17 @@ def signature(self) -> Signature:
     def from_qsp_polynomial(
         cls,
         U: GateWithRegisters,
-        P: Sequence[complex],
+        P: Union[Sequence[complex], Shaped],
         *,
         negative_power: int = 0,
         verify: bool = False,
         verify_precision=1e-7,
     ) -> 'GeneralizedQSP':
+        import cirq
+
+        if cirq.is_parameterized(P):
```

Comment:
do we need cirq for this?

Reply:
actually no, I updated the `is_symbolic` function to mirror the implementation of `cirq.is_parameterized`.

Reply:
sweeeeeeeeet